types/decimal_test.go

```diff
@@ -0,0 +1,69 @@
+package types_test
+
+import (
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestDecimal(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
```

Comment:
Could you swap `name` and `f` to avoid the field alignment warning?

Reply:
Could you also add a test for scale out of bounds?

Reply:
`big.Int` already prevents out-of-bound input (e.g. `big.NewInt(math.MaxInt64+1)`), so it's only possible for scale value.

FYI, we can also restrict `scale` to 32-bit size by changing the parameter to `int32` from `int`. 

Java implementation does not restrict both of the inputs. Currently out-of-bound `scale` values works just fine similar to Java implementation.